kenfeliciano/koamar.com
# Update to gatsby 4

This PR *finally* upgrades the site from Gatsby 2 to Gatsby 4. I also upgraded Tailwind CSS to v3.
- `npm install @latest` all the things
- rollback update to `@mdx-js/mdx` and `@mdx-js/react` because `gatsby-plugin-mdx` requires v1
- uninstall `gatsby-image`
- install `gatsby-plugin-image`
- run codemod for `gatsby-plugin-image` to fix a lot of things automatically
- change purge to content in tailwind.config.js
- make manual changes required for `gastby-plugin-image`
- remove Image component and just use `StaticImage` from `gatsby-plugin-image` for the one component that has a static image

Shockingly, it all worked after all was done. 

Before this, I was seeing strange space taken up from what I believed at the time, was SVG images being their default size for a moment before shrinking to the size specified. Meanwhile, they would leave the space they took at their default size. This *seems* to have stopped. I did not see it locally with either `gatsby develop` or `gatsby build` so I'm hopeful!
